lib/ramble/ramble/pipeline.py

```diff
@@ -521,9 +512,136 @@ def _execute(self):
             executor(*exec_args)
 
 
+class PublishDeploymentPipeline(Pipeline):
+    """class for the `prepare-deployment` pipeline"""
+
+    name = 'publishdeployment'
+    index_filename = 'index.json'
+    index_namespace = 'deployment_files'
+    tar_extension = '.tar.gz'
+
+    def __init__(self, workspace, filters,
+                 create_tar=False, upload_url=None,
+                 deployment_name=None):
+        super().__init__(workspace, filters)
+        self.action_string = 'Publishing deployment of'
+        self.require_inventory = True
+        self.create_tar = create_tar
+        self.upload_url = upload_url
+        self.object_repo_name = 'object_repo'
+
+        if deployment_name:
+            workspace.deployment_name = deployment_name
+            self.deployment_name = deployment_name
+        else:
+            self.deployment_name = workspace.name
+
+    def _execute(self):
+        configs_dir = os.path.join(self.workspace.named_deployment,
+                                   ramble.workspace.workspace_config_path)
+        fs.mkdirp(configs_dir)
+
+        _copy_tree(self.workspace.config_dir, configs_dir)
+
+        aux_software_dir = os.path.join(configs_dir, ramble.workspace.auxiliary_software_dir_name)
+        fs.mkdirp(aux_software_dir)
+        aux_repo_conf = os.path.join(aux_software_dir, 'repos.yaml')
+        if os.path.exists(aux_repo_conf):
+            f = open(aux_repo_conf, 'a')
+        else:
+            f = open(aux_repo_conf, 'w+')
+            f.write('repos:\n')
+        f.write(f'- ../../{self.object_repo_name}\n')
```

Comment:
There is some bug where if you do a `publish` for the same workspace we instead keep appending to files and end up with an invalid distribution. 

First publish:

```
./deployments/gke/configs/modifier_repos.yaml:- ../object_repo
./deployments/gke/configs/repos.yaml:- ../object_repo
```

After many publishes:
```
 $ grep -r "object_repo" .
./deployments/gke/configs/modifier_repos.yaml:- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo
./deployments/gke/configs/repos.yaml:- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo- ../object_repo
./deployments/gke/configs/auxiliary_software_files/repos.yaml:- ../../object_repo
./deployments/gke/configs/auxiliary_software_files/repos.yaml:- ../../object_repo
./deployments/gke/configs/auxiliary_software_files/repos.yaml:- ../../object_repo
./deployments/gke/configs/auxiliary_software_files/repos.yaml:- ../../object_repo
./deployments/gke/configs/auxiliary_software_files/repos.yaml:- ../../object_repo
```
I think we need to add some logic around cleaning out old deployments

Reply:
Ah that's fun. OK, I can look at that.